flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/serde/LogicalTypeJsonSerializer.java

```diff
@@ -87,6 +89,10 @@ public final class LogicalTypeJsonSerializer extends StdSerializer<LogicalType>
     public static final String FIELD_NAME_CLASS = "class";
     public static final String FIELD_NAME_SERIALIZER = "serializer";
 
+    // INTERVAL
```

Comment:
I just wonder whether it's better to serialize the INTERVAL type as follows:

```
{
    "type": "INTERVAL",
    "upperResolution": {
          "precision": ...,
          "intervalUnit": "..."
    },
    "lowerResolution": ...
}
```

So we can align with the API in `DataTypes#INTERVAL`. WDYT? @twalthr @lincoln-lil

Reply:
+1 for the alignment with datatype api, it can be easier to understand

Reply:
From one side I'm ok to add `upper/lowerResolution`
however from another side current ser/de is based on `LogicalTypeRoot`. And `INTERVAL` is NOT a `LogicalTypeRoot`... it is just a `LogicalTypeFamily`. So it will require extra parsing to differentiate between `YearMonthIntervalType` and `DayTimeIntervalType`.
Or am I missing something here?

Reply:
`INTERVAL` is a kind of `DataType` in the doc[1]. I think the API here aims at the DataType rather than LogicalType in the design[2].

[1] https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/types/#list-of-data-types
[2] https://cwiki.apache.org/confluence/display/FLINK/FLIP-91%3A+Support+SQL+Gateway#FLIP91:SupportSQLGateway-SerializeanddeserializetheResultSet

Reply:
updated usage of `precision` with usage of `yearPrecision` and `dayPrecision` based on offline discussion in slack